### PR TITLE
parser: add generated column grammar.

### PR DESCRIPTION
**Pull request by hicqu**
add reserved keywords: generated, virtual, stored.
add nonreserved keywords: always.

### CODE REVIEW

**Comment by hicqu** [comment]
@tiancaiamao @XuHuaiyu @coocood , PTAL, thanks.

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
Why CI failed?
Rest LGTM
@hicqu

**Comment by hicqu** [comment]
all CI are passed. @XuHuaiyu

File: parser/parser.y [hunk 1]
```diff
@@ -1297,6 +1302,34 @@ ColumnOption:
 		// The CHECK clause is parsed but ignored by all storage engines.
 		$$ = &ast.ColumnOption{}
 	}
+|	GeneratedAlways "AS" '(' Expression ')' VirtualOrStored
+	{
+		startOffset := parser.startOffset(&yyS[yypt-2])
+		endOffset := parser.endOffset(&yyS[yypt-1])
+		expr := $4.(ast.ExprNode)
+		expr.SetText(parser.src[startOffset:endOffset])
```
Comment (tiancaiamao) [inline parser/parser.y]: Test the result text is as you wished.

File: parser/misc.go [hunk 2]
```diff
@@ -146,6 +146,7 @@ var tokenMap = map[string]int{
 	"AFTER":                      after,
 	"ALL":                        all,
 	"ALTER":                      alter,
+	"ALWAYS":                     always,
```
Comment (tiancaiamao) [inline parser/misc.go]: keyword should be add to `ReservedKeyword` or `UnReservedKeyword`, you can find it searching
`^ReservedKeyword` in parser.y , and add them to TestSimple in parser_test.go

Comment (hicqu) [inline parser/misc.go]: it's already in unreserved keywords list.

File: parser/parser.y (hunk 1 above)
Comment (hicqu) [inline parser/parser.y]: fixed.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**